PROTO_tests/tests/pdarray_creation_test.py

```diff
@@ -0,0 +1,742 @@
+            assert isinstance(pda, ak.pdarray)
+            assert 10 == len(pda)
+            assert ak.bigint == pda.dtype
+            assert pda[-1] == bi + 10 - 1
+
+        # test array and arange infer dtype
+        assert (
+            ak.array([bi, bi + 1, bi + 2, bi + 3, bi + 4]).to_list() == ak.arange(bi, bi + 5).to_list()
+        )
+
+        # test that max_bits being set results in a mod
+        assert ak.arange(bi, bi + 5, max_bits=200).to_list() == ak.arange(5).to_list()
+
+        # test ak.bigint_from_uint_arrays
+        # top bits are all 1 which should be 2**64
+        top_bits = ak.ones(5, ak.uint64)
+        bot_bits = ak.arange(5, dtype=ak.uint64)
+        two_arrays = ak.bigint_from_uint_arrays([top_bits, bot_bits])
+        assert ak.bigint == two_arrays.dtype
+        assert two_arrays.to_list() == [2**64 + i for i in range(5)]
+        # top bits should represent 2**128
+        mid_bits = ak.zeros(5, ak.uint64)
+        three_arrays = ak.bigint_from_uint_arrays([top_bits, mid_bits, bot_bits])
+        assert three_arrays.to_list() == [2**128 + i for i in range(5)]
+
+        # test round_trip of ak.bigint_to/from_uint_arrays
+        t = ak.arange(bi - 1, bi + 9)
+        t_dup = ak.bigint_from_uint_arrays(t.bigint_to_uint_arrays())
+        assert t.to_list() == t_dup.to_list()
+        assert t_dup.max_bits == -1
+
+        # test setting max_bits after creation still mods
+        t_dup.max_bits = 200
+        assert t_dup.to_list() == [bi - 1, 0, 1, 2, 3, 4, 5, 6, 7, 8]
+
+        # test slice_bits along 64 bit boundaries matches return from bigint_to_uint_arrays
+        for i, uint_bits in enumerate(t.bigint_to_uint_arrays()):
+            slice_bits = t.slice_bits(64 * (4 - (i + 1)), 64 * (4 - i) - 1)
+            assert uint_bits.to_list() == slice_bits.to_list()
+
+    def test_arange(self):
+        assert np.arange(0, 10, 1).tolist() == ak.arange(0, 10, 1).to_list()
+        assert np.arange(10, 0, -1).tolist() == ak.arange(10, 0, -1).to_list()
+        assert np.arange(-5, -10, -1).tolist() == ak.arange(-5, -10, -1).to_list()
+        assert np.arange(0, 10, 2).tolist() == ak.arange(0, 10, 2).to_list()
+
+    @pytest.mark.parametrize("dtype", ak.intTypes)
+    def test_arange_dtype(self, dtype):
+        # test dtype works with optional start/stride
+        stop = ak.arange(100, dtype=dtype)
+        assert np.arange(100, dtype=dtype).tolist() == stop.to_list()
+        assert dtype == stop.dtype
+
+        start_stop = ak.arange(100, 105, dtype=dtype)
+        assert np.arange(100, 105, dtype=dtype).tolist() == start_stop.to_list()
+        assert dtype == start_stop.dtype
+
+        start_stop_stride = ak.arange(100, 105, 2, dtype=dtype)
+        assert np.arange(100, 105, 2, dtype=dtype).tolist() == start_stop_stride.to_list()
+        assert dtype == start_stop_stride.dtype
+
+    def test_arange_misc(self):
+        # test uint64 handles negatives correctly
```

Comment:
We should add this one since it makes sure we correctly return a `uint` for values in the `2**63..2**64` range
```python
uint_start_stop = ak.arange(2**63 + 3, 2**63 + 7)
ans = ak.arange(3, 7, dtype=ak.uint64) + 2**63
self.assertListEqual(ans.to_list(), uint_start_stop.to_list())
self.assertEqual(ak.uint64, uint_start_stop.dtype)
```